elements/rh-video/rh-video.ts

```diff
@@ -0,0 +1,258 @@
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators/property.js';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { state } from 'lit/decorators/state.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { SlotController } from '@patternfly/pfe-core/controllers/slot-controller.js';
+import '../rh-button/rh-button.js';
+import '../rh-surface/rh-surface.js';
+
+import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
+
+import styles from './rh-video.css';
+
+export class ConsentClickEvent extends Event {
+  declare target: RhVideo;
+  constructor() {
+    super('consent-click', { bubbles: true, cancelable: true });
+  }
+}
+
+export class VideoClickEvent extends Event {
+  declare target: RhVideo;
+  constructor() {
+    super('play-button-click', { bubbles: true, cancelable: true });
```

Comment:
i'm not sure about this event name... it's not clear to me why we need both this and `play`. why not just cancel the play event?

Reply:
Good question.

It could be because:

1. Page loads, thumbnail image + `<button>` are visible on the page (note: YouTube embed is _not_ on the page at this point)
1. A user clicks the play button and the `play-button-click` fires.
1. The thumbnail image + `<button>` are hidden and the YouTube embed loads on the page.
1. The video starts playing.
1. `play` fires.

It seems plausible that someone might:

1. Want to have an event for when someone clicks the button and
1. Have another event for after the video starts playing.

In my testing, hooking into the YouTube event API when the `<iframe>` doesn't exist on page load is a challenge. The `play-button-clicked` and `play` events could help our users here.

Reply:
`'request-play``
`'stream'`
`'request-load'`

Reply:
Resolved in https://github.com/RedHat-UX/red-hat-design-system/pull/1514/commits/e32a4c0afa62351e359e09549fe23d3d6dfdf573.